camel/toolkits/web_toolkit.py

```diff
@@ -0,0 +1,321 @@
+import os
+import subprocess
+import json
+from typing import Any, Dict, List
+
+from camel.toolkits.base import BaseToolkit
+from camel.toolkits.function_tool import FunctionTool
+from camel.agents import ChatAgent
+from camel.models import ModelFactory
+from camel.types import ModelPlatformType, ModelType
+from camel.configs import ChatGPTConfig
+from camel.messages import BaseMessage
+
+
+class WebToolkit(BaseToolkit):
+    r"""A class representing a toolkit for web use.
+
+    This class provides methods for interacting with websites by writing direct JavaScript code via tools like Stagehand.
+    """
+
+    def stagehand_tool(self, task_prompt: str) -> Dict[str, Any]:
+        r"""
+        Single entry point that:
+         1) Generates Stagehand JavaScript code to interact with the web
+         2) Executes it under Node.js
+         3) Returns the final JSON result
+
+        Args:
+            task_prompt (str): Description of the task to automate.
+
+        Returns:
+            Dict[str, Any]: JSON result from the Stagehand script, or an error.
+        """
+
+        # Generate Stagehand code
+        js_code = self._generate_stagehand_code(task_prompt)
+
+        # Run code in Node, capture JSON
+        result_str = self._run_stagehand_script_in_node(js_code)
+
+        # Attempt to parse JSON
+        try:
+            return json.loads(result_str)
+        except json.JSONDecodeError:
+            return {
+                "status": "error",
+                "message": f"No valid JSON output. Last script line:\n{result_str}",
+            }
+
+    #
+    # Internals
+    #
+    def _generate_stagehand_code(self, high_level_task: str) -> str:
+        r"""
+        Internal method for generating Stagehand code.
+        """
+        model = ModelFactory.create(
+            model_platform=ModelPlatformType.OPENAI,
+            model_type=ModelType.GPT_4O_MINI,
+            model_config_dict=ChatGPTConfig(temperature=0.0).as_dict(),
+        )
+
+        # A system message to instruct how to generate Stagehand code
+        agent = ChatAgent(
+            BaseMessage.make_assistant_message(
```

Comment:
`BaseMessage.make_assistant_message` maybe depreciated soon. We can just use raw `string` here.